evalml/tests/automl_tests/test_default_algorithm.py

```diff
@@ -264,3 +308,76 @@ def test_default_algo_drop_columns(mock_get_names, columns, X_y_binary):
                     pipeline.parameters["Drop Columns Transformer"]["columns"]
                     == columns
                 )
+
+    batch = algo.next_batch()
+    add_result(algo, batch)
+    for pipeline in batch:
+        for component_name in pipeline.component_graph.compute_order:
+            split = component_name.split(" - ")
+            if "Drop Columns Transformer" in split:
+                assert algo._pipeline_params[component_name]["columns"] == columns
+                assert pipeline.parameters[component_name]["columns"] == columns
+
+
+def test_make_split_pipeline(X_y_binary):
+    X, y = X_y_binary
+    algo = DefaultAlgorithm(X, y, ProblemTypes.BINARY, sampler_name=None)
+    algo._selected_cols = ["1", "2", "3"]
+    algo._selected_cat_cols = ["A", "B", "C"]
+    pipeline = algo._make_split_pipeline(RandomForestClassifier, "test_pipeline")
+    compute_order = [
+        "Label Encoder",
+        "Categorical Pipeline - Select Columns Transformer",
+        "Categorical Pipeline - Label Encoder",
+        "Categorical Pipeline - Imputer",
+        "Numeric Pipeline - Label Encoder",
+        "Numeric Pipeline - Imputer",
+        "Numeric Pipeline - Select Columns Transformer",
+        "Random Forest Classifier",
+    ]
+    assert pipeline.component_graph.compute_order == compute_order
+    assert pipeline.name == "test_pipeline"
+    assert pipeline.parameters["Numeric Pipeline - Select Columns Transformer"][
+        "columns"
+    ] == ["1", "2", "3"]
+    assert pipeline.parameters["Categorical Pipeline - Select Columns Transformer"][
+        "columns"
+    ] == ["A", "B", "C"]
+    assert isinstance(pipeline.estimator, RandomForestClassifier)
+
+
+@patch("evalml.pipelines.components.FeatureSelector.get_names")
+@patch("evalml.pipelines.components.OneHotEncoder._get_feature_provenance")
+def test_select_cat_cols(
+    mock_get_feature_provenance, mock_get_names, X_y_categorical_classification
+):
+    X, y = X_y_categorical_classification
+    X.ww.init()
+    cat_cols = list(X.ww.select("categorical").columns)
+    mock_get_names.return_value = ["0", "1", "2", "Sex_male", "Embarked_S"]
+    mock_get_feature_provenance.return_value = {
+        "Sex": ["Sex_male"],
+        "Embarked": ["Embarked_S"],
+    }
+
+    algo = DefaultAlgorithm(X, y, ProblemTypes.BINARY, None)
+
+    batch = algo.next_batch()
+    add_result(algo, batch)
+
+    batch = algo.next_batch()
+    add_result(algo, batch)
```

Comment:
do we need to do 2 calls for this?

Reply:
yes, its to get it to the correct batch!

Reply:
Can you add a comment on that for future reference? I'm sure i'll forget lol